liesl/streams/mock.py

```diff
@@ -171,28 +176,33 @@ def run(self):
         self.is_running = True
         timestamps = self.stream['time_stamps']
         timeseries = self.stream['time_series']
+        num_of_samples = len(timestamps)
 
         while self.is_running:
-            smpl = timeseries[count]
+            smpl = timeseries[count % num_of_samples]
             count += 1
             # now send it and wait for a bit
             outlet.push_sample(smpl)
-            time.sleep(timestamps[count+1]-timestamps[count])
+            time_diff = timestamps[(count + 1) % num_of_samples] - timestamps[count % num_of_samples]
+            if time_diff > 0:
+                time.sleep(time_diff)
+            else:
+                time.sleep(0.001)
 
```

Comment:
The sleep in line 188/190 exhibits a bias, as you do not account for time spent in lines 182 to 186. One option is to measure this time spent, and reduce the time slept by this time, see re.g. Clock.sleep_debiased from Pyreiz:

https://github.com/pyreiz/pyreiz/blob/master/reiz/time.py#L83-L93

Reply:
Also, sleep has an inherent positive bias, that adds up over repeated calls. Consider a self-correcting scheduler implementation, e.g. https://gist.github.com/agricolab/1f35cecea25c3e251d40057f2221f831#file-periodic_self_correcting-py